lib/Room.php

```diff
@@ -550,7 +572,9 @@ public function removeParticipantBySession(Participant $participant) {
 	 * @throws InvalidPasswordException
 	 */
 	public function joinRoom($userId, $password, $passedPasswordProtection = false) {
-		$this->dispatcher->dispatch(self::class . '::preJoinRoom', new GenericEvent($this));
+		$this->dispatcher->dispatch(self::class . '::preJoinRoom', new GenericEvent($this, [
+			'userId' => $userId,
```

Comment:
for consistancy add the password and the flag too and also add both parameters at the postJoinRoom event below